src/seq/mod.rs

```diff
@@ -191,17 +191,7 @@ pub trait IteratorRandom: Iterator + Sized {
         let mut result = None;
 
         if upper == Some(lower) {
-            // Remove this once we can specialize on ExactSizeIterator
             return if lower == 0 { None } else { self.nth(rng.gen_range(0, lower)) };
```

Comment:
Yeah, removing the comment is probably a good idea. See for example this for a good reason we probably will want to keep this code around no matter what: https://github.com/rust-lang/rust/issues/34433